public/js/notifications.js

```diff
@@ -0,0 +1,433 @@
+                    + "this browser does not support the 'Notification API' in the"
+                    + " current context"
+                );
+                isValidated = false;
+            }
+            if (! isValidated) {
+                // we only log the error and exit early as throwing would completely hang up the web application
+                this.logger.error("The 'Notification' module is missing some required API's");
+                return;
+            }
+
+            this.logger.debug(this.prefix + "spawned");
+            this.load();
+        }
+
+        load () {
+            this.logger.debug(this.prefix + "loading");
+
+            // listen to render events on container for col1 (to inject notification toggle)
+            this.on('rendered', '#main > #col1.container', this.renderHandler, this);
+
+            // listen to controller (service worker) changes
+            navigator.serviceWorker.addEventListener('controllerchange', (event) => {
+                this.logger.debug(this.prefix + "new controller attached ", event.target.controller);
+                if (event.target.controller !== null) {
+                    // reset eventsource and handshake flag
+                    this.allowedToOperate = false;
+                    if (this.eventSource !== null && this.eventSource.readyState !== EventSource.CLOSED) {
+                        this.eventSource.close();
+                    }
+
+                    this.logger.debug(this.prefix + "send handshake to controller");
+                    event.target.controller.postMessage(
+                        JSON.stringify({
+                            command: 'handshake',
+                            version: VERSION
+                        })
+                    );
+                }
+            });
+
+            // listen to messages from the controller (service worker)
+            self.navigator.serviceWorker.addEventListener('message', (event) => {
+                if (event.data) {
+                    let data = JSON.parse(event.data);
+                    switch (data.command) {
+                        case 'handshake':
+                            if (data.status === 'outdated') {
+                                this.logger.debug(
+                                    this.prefix
+                                    + "handshake got rejected as we're running an outdated script version"
+                                );
+
+                                // the controller declared us as an outdated script version
+                                this.icinga.loader.createNotice(
+                                    'warning',
+                                    'This tab is running an outdated script version. Please reload the page!',
+                                    true
+                                );
+
+                                this.allowedToOperate = false;
+                            } else if (data.status === 'accepted') {
```

Comment:
The condition seems to be superfluous here.

Reply:
This has been done on purpose, as in the future other status types could be introduced by the service worker. We currently only check for the tab version validity, but it's technically not limited to that.
As an example, the service worker could accept the handshake but with additional requirements.